### PR TITLE
fix: hotkey swap delegates

### PR DESCRIPTION
**Pull request by distributedstatemachine**
## Description

This PR fixes a bug with the hotkey swap , where a hot key automatically becomes a delegate after a hotkey swap.

### CODE REVIEW

File: pallets/subtensor/tests/swap_hotkey.rs [hunk 1]
```diff
@@ -959,3 +959,38 @@ fn test_swap_hotkey_error_cases() {
...
+        assert!(!Delegates::<Test>::contains_key(old_hotkey));
+
+        // Perform the swap
+        assert_ok!(SubtensorModule::do_swap_hotkey(
+            <<Test as Config>::RuntimeOrigin>::signed(coldkey),
+            &old_hotkey,
+            &new_hotkey
+        ));
+
+        // Check that old_hotkey is no longer a delegate
```
Comment (camfairchild) [inline pallets/subtensor/tests/swap_hotkey.rs]:
```suggestion
        // Check that old_hotkey is still not a delegate
```

File: pallets/subtensor/tests/swap_hotkey.rs [hunk 2]
```diff
@@ -959,3 +959,38 @@ fn test_swap_hotkey_error_cases() {
...
+        assert!(!Delegates::<Test>::contains_key(old_hotkey));
+
+        // Check that new_hotkey is now a delegate with the correct take value
```
Comment (camfairchild) [inline pallets/subtensor/tests/swap_hotkey.rs]:
```suggestion
        // Check that new_hotkey is NOT a delegate either
```

File: pallets/subtensor/tests/swap_hotkey.rs [hunk 3]
```diff
@@ -959,3 +959,38 @@ fn test_swap_hotkey_error_cases() {
         assert_eq!(Balances::free_balance(coldkey), initial_balance - swap_cost);
     });
 }
+
+// SKIP_WASM_BUILD=1 RUST_LOG=debug cargo test --test swap_hotkey -- test_swap_hotkey_becomes_delegate --exact --nocapture
+#[test]
+fn test_swap_hotkey_becomes_delegate() {
```
Comment (camfairchild) [inline pallets/subtensor/tests/swap_hotkey.rs]:
```suggestion
// SKIP_WASM_BUILD=1 RUST_LOG=debug cargo test --test swap_hotkey -- test_swap_hotkey_does_not_become_delegate --exact --nocapture
#[test]
fn test_swap_hotkey_does_not_become_delegate() {
```

**Review by camfairchild — DISMISSED** [review]
See suggestions

**Review by camfairchild — APPROVED** [review]

**Review by JohnReedV — APPROVED** [review]

**Pull request merged**